### ISSUE TITLE
let_underscore_untyped false positive when using _ as a parameter name in an async function

### ISSUE THREAD

**Issue by laralove143** [rust-lang/rust-clippy#12395]
### Summary

when naming a parameter `_` in an async function, the lint throws a false-positive

### Lint Name

let_underscore_untyped

### Reproducer

I tried this code:

```rust
async fn foo(_: u8) {}
```

I saw this happen:

```
warning: non-binding `let` without a type annotation
   --> src/main.rs:1:14
    |
115 | async fn foo(_: u8) {}
    |              ^
    |
help: consider adding a type annotation
   --> src/main.rs:1:15
    |
115 | async fn foo(_: u8) {}
    |               ^
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#let_underscore_untyped
```

I expected to see this happen:
no warning

### Version

```text
rustc -Vv
rustc 1.76.0 (07dca489a 2024-02-04)
binary: rustc
commit-hash: 07dca489ac2d933c78d3c5158e3f43beefeb02ce
commit-date: 2024-02-04
host: aarch64-apple-darwin
release: 1.76.0
LLVM version: 17.0.6
```
